lib/prometheus_exporter/middleware.rb

```diff
@@ -6,12 +6,38 @@
 class PrometheusExporter::Middleware
   MethodProfiler = PrometheusExporter::Instrumentation::MethodProfiler
 
+  module RedisInstrumenter
+    def call(_command, _config)
+      unless prof = Thread.current[:_method_profiler]
+        super
```

Comment:
Don't you end up calling `super` twice here? That would emit the redis command twice.

Reply:
oh implementation bug, will fix and add a test